lib/Graphs/ICFG.cpp

```diff
@@ -460,11 +472,16 @@ struct DOTGraphTraits<ICFG*> : public DOTGraphTraits<PAG*>
         if (IntraBlockNode* bNode = SVFUtil::dyn_cast<IntraBlockNode>(node))
         {
             rawstr << "IntraBlockNode ID: " << bNode->getId() << " \t";
-            PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
-            for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
-            {
-                const PAGEdge* edge = *it;
-                rawstr << edge->toString();
+            rawstr << value2String(bNode->getInst()) << " \t";
+            if (Options::IncludePAGInICFGDump) {
+                if (PAG* pag = PAG::tryGetPAG()) {
```

Comment:
This branch condition will never be false since `PAG::tryGetPAG()` will always return a PAG.

Reply:
But if the PAG has not yet been created, what it will return is null, so it must be checked. That could happen during the window of time (mentioned above), between when the ICFG is created and the PAG is made.

Reply:
The `tryGetPAG ` is risky once exposed to users and might cause a null pointer. PAG is the most fundamental graph in SVF and it should always be available. 

I do not think we have a window here. The ICFG once built, the PAG should always available.  Do you mean someone may try to create an ICFG instance and then try to dump it inside the PAG constructor?
https://github.com/SVF-tools/SVF/blob/master/lib/Graphs/PAG.cpp#L265

If that is the case, I think we should consider making ICFG's constructor private and make the whole analysis share one ICFG similar to PAG.